solver/src/settlement_submission.rs

```diff
@@ -41,9 +51,7 @@ pub async fn submit(
         .context("failed to get pending gas price")?;
 
     // Account for some buffer in the gas limit in case racing state changes result in slightly more heavy computation at execution time
-    let gas_limit = retry::settle_method_builder(contract, settlement.clone())
-        .tx
-        .estimate_gas()
+    let gas_limit = estimate_gas(contract, &settlement)
```

Comment:
nanonit: if submit took a RatedSettlement it could reuse the gas estimate from that struct instead of estimating it again.